Assets/Tests/EditMode/AuthTests/GeneralAccessTokenProvidersStrategyTests.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using Moq;
+using NUnit.Framework;
+using Src.AuthController;
+
+namespace Tests.EditMode.AuthTests
+{
+    public class GeneralAccessTokenProvidersStrategyTests
+    {
+        [Test]
+        [TestCaseSource(nameof(GetAuthTypes))]
+        public void GetAccessTokenProvider_ShouldReturnCorrectTokenProvider_ObtainedFromStrategy(AuthType authType)
+        {
+            var expectedFirebaseTokenProvider = new FirebaseTokenProvider();
```

Comment:
Consider creating static methods GetFirebaseTokenProvider() and GetMetamaskTokenProvider() in ObjectCreationUtility class, as constructors of these providers may change in the future and it would be unpleasant to update all the dependent tests because of it.